usefulUtilities/avatarEntityRemover/avatarEntityRemover.js

```diff
@@ -0,0 +1,155 @@
+        var props = Entities.getEntityProperties(entityID, ['avatarEntity', 'entityHostType', 'locked', 'owningAvatarID', 'collisionless']);
+        
+        if (props.owningAvatarID === MyAvatar.sessionUUID && (props.avatarEntity || props.entityHostType === "avatar")) {
+
+            if (enableCollisionlessAvatarEntities && props.collisionless) {
+                return;
+            }
+
+            if (props.locked) {
+                console.log("Boot Code 00000002");
+                Window.location = kickDomain;
+                return;
+            }
+
+            // This will do nothing unless the user has lock/unlock rights
+            Entities.editEntity(entityID, {locked: false});
+
+            Entities.editEntity(entityID, {"dimensions": {"x": 0.01, "y": 0.01, "z": 0.01}});
+            Entities.deleteEntity(entityID);
+        }
+    }
+
+
+    function maybeRemoveAllCurrentAvatarEntities() {
+        if (APPROVED_USERNAMES.indexOf(AccountServices.username) > -1) {
+            return;
+        }
+
+        MyAvatar.getAvatarEntitiesVariant().forEach(function(avatarEntity) {
+            if (enableCollisionlessAvatarEntities && avatarEntity.properties.collisionless) {
+                return;
+            }
+
+            Entities.deleteEntity(avatarEntity.id);
+        });
+    }
+
+
+    function maybeSaveThenRemoveAllCurrentAvatarEntities() {
+        if (APPROVED_USERNAMES.indexOf(AccountServices.username) > -1) {
+            return;
+        }
+
+        MyAvatar.getAvatarEntitiesVariant().forEach(function(avatarEntity) {
+            if (avatarEntity.properties.locked) {
+                console.log("Boot Code 00000001");
+                Window.location = kickDomain;
+                return;
+            }
+
+            if (enableCollisionlessAvatarEntities && avatarEntity.properties.collisionless) {
+                return;
+            }
+
+            maybeSaveThenDelete(avatarEntity.id, avatarEntity.properties);
+        });
+    }
+    
+
+    var AvatarEntityRemover = function() {};
+
+    AvatarEntityRemover.prototype = {
+        preload: function (id) {
+            var properties = Entities.getEntityProperties(id, ["userData"]);
+            var userData;
+
+            try {
+                userData = JSON.parse(properties.userData);
+            } catch (e) {
+                console.error("Error parsing userData: ", e);
+            }
+
+            if (userData) {
+                if (userData.enableAvatarEntityRestore) {
+                    enableAvatarEntityRestore = userData.enableAvatarEntityRestore;
+                }
+
+                if (userData.kickDomain) {
+                    kickDomain = userData.kickDomain;
+                }
+
+                if (userData.enableCollisionlessAvatarEntities) {
+                    enableCollisionlessAvatarEntities = userData.enableCollisionlessAvatarEntities;
+                }
+            }
+
+            Entities.addingEntity.connect(onAddingEntity);
```

Comment:
on interesting, avatar entities give this signal also?

Reply:
yes! otherwise i'd be out of luck :D